modules/widget_utils.js

```diff
@@ -129,19 +131,41 @@ exports.swipeOn = function(autohide) {
   function anim(dir, callback) {
     if (exports.animInterval) clearInterval(exports.animInterval);
     exports.animInterval = setInterval(function() {
-      exports.offset += dir;
+// exports.offset < -23 + |dir|     > -|dir|         otherwise
+//     dir >0     start showing     shown this step  showing
+//         <0     hidden this step  start hiding     hiding
       let stop = false;
-      if (dir>0 && exports.offset>=0) { // fully down
-        stop = true;
-        exports.offset = 0;
-      } else if (dir<0 && exports.offset<-23) { // fully up
-        stop = true;
-        exports.offset = -24;
+      if (dir > 0) {
+        if (exports.offset >= -dir) {
+          // nearly shown
+          stop = true;
+          exports.offset = 0; // clamp
+          Bangle.emit("widgets-shown");
+        } else if (exports.offset < -23 + dir) {
+          Bangle.emit("widgets-start-show");
+        } else {
+          Bangle.emit("widgets-anim-step");
+        }
```

Comment:
Do we need these changes? clkinfo is loaded by a lot of apps so it has a high barrier to additional complexity/code. For instance, `widgets-anim-step` isn't used